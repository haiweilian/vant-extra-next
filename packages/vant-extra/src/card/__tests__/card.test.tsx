import { mount } from '@vue/test-utils'
import { describe, test, expect } from 'vitest'
import { Card } from '../index'

describe('Card.vue', () => {
  test('test component props', async () => {
    const wrapper = mount(() => (
      <Card
        shadow
        title="This is title"
        desc="This is desc"
        src="https://fastly.jsdelivr.net/npm/@vant/assets/cat.jpeg"
        fit="cover"
      />
    ))

    expect(wrapper.find('.vae-card').classes('vae-card--shadow')).toBe(true)

    expect(wrapper.find('.vae-card__title').text()).toBe('This is title')

    expect(wrapper.find('.vae-card__desc').text()).toBe('This is desc')
  })

  test('test component slots', async () => {
    const slots = {
      extra: () => 'This is extra',
      default: () => 'This is content',
    }
    const wrapper = mount(() => (
      <Card
        shadow
        title="This is title"
        desc="This is desc"
        src="https://fastly.jsdelivr.net/npm/@vant/assets/cat.jpeg"
        fit="cover"
        v-slots={slots}
      />
    ))

    expect(wrapper.html()).toMatchSnapshot()
  })
})
