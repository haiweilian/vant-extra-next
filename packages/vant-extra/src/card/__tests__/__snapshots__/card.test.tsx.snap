// Vitest Snapshot v1

exports[`Card.vue > test component slots 1`] = `
"<div class=\\"vae-card vae-card--shadow\\">
  <div class=\\"van-image vae-card__image\\"><img src=\\"https://fastly.jsdelivr.net/npm/@vant/assets/cat.jpeg\\" class=\\"van-image__img\\" style=\\"object-fit: cover;\\">
    <div class=\\"van-image__loading\\"><i class=\\"van-badge__wrapper van-icon van-icon-photo van-image__loading-icon\\">
        <!---->
        <!---->
        <!---->
      </i></div>
    <!---->
  </div>
  <div class=\\"vae-card__title\\">This is title</div>
  <div class=\\"vae-card__desc\\">This is desc</div>This is content<div class=\\"vae-card__extra\\">This is extra</div>
</div>"
`;
